core/src/main/java/org/elasticsearch/index/reindex/WorkerBulkByScrollTaskState.java

```diff
@@ -189,7 +195,8 @@ public void delayPrepareBulkRequest(ThreadPool threadPool, TimeValue lastBatchSt
 
     public TimeValue throttleWaitTime(TimeValue lastBatchStartTime, TimeValue now, int lastBatchSize) {
         long earliestNextBatchStartTime = now.nanos() + (long) perfectlyThrottledBatchTime(lastBatchSize);
-        return timeValueNanos(max(0, earliestNextBatchStartTime - System.nanoTime()));
+        long waitTime = min(MAX_THROTTLE_WAIT_TIME, max(0, earliestNextBatchStartTime - System.nanoTime()));
```

Comment:
I like limiting this here but think maybe we ought to also enforce it at request start and rethrottle time. We can work backwards from the batch size to reject requests that had a requests_per_second that is too small, I think.

Reply:
I wonder if we should just do that in a followup.

Reply:
I think it might be nice to use the actual value of the setting but that can be a "reindex only" followup thing, I think.